include/ck_tile/core/utility/reduce_operator.hpp

```diff
@@ -85,6 +85,22 @@ struct Max
     {
         return max(y, x);
     }
+
+    // Overload with changed flag for index tracking
+    template <typename T,
+              typename = std::enable_if_t<std::is_same_v<T, float> || std::is_same_v<T, double> ||
+                                          std::is_same_v<T, int32_t> || std::is_same_v<T, int8_t> ||
+                                          std::is_same_v<T, half_t> || std::is_same_v<T, bf16_t> ||
+                                          std::is_same_v<T, fp8_t> || std::is_same_v<T, bf8_t>>>
```

Comment:
We have a utility function like `is_any_of<....>` which would make this code much more concise.